### PR TITLE
[embree3] does supports arm64-osx

### LINKED ISSUE #30541: [embree3] Enable Mac Arm
****
Intel® Embree is a high-performance ray tracing library developed at Intel, which is released as open source under the [Apache 2.0 license](http://www.apache.org/licenses/LICENSE-2.0). Intel® Embree supports x86 CPUs under Linux, macOS, and Windows; **_ARM CPUs on macOS;_** as well as Intel® GPUs under Linux and Windows.

**Is your feature request related to a problem? Please describe.**
I'd like to use embree on mac arm : https://github.com/embree/embree

**Proposed solution**
Suggested in vcpkg.json : 
  "supports": "!arm" =>   "supports": "!(arm & (windows | linux))"
But would also need to change the default feature list : On arm those features are not available
"default-features": [
    "avx",
    "avx2",
    "sse2",
    "sse42"
  ],

**I'm not sure how safe it is to remove the default features** ?
Also the port can be updated to 4.0.1 but than can be done after. 
**Describe alternatives you've considered**
A clear and concise description of any alternative solutions or features you've considered if applicable.

**Additional context**
I've applied the change in vcpkg.json (Supports and removed default features). Then the build seem to be OK : 
cpkg % ./vcpkg install embree3
Computing installation plan...
The following packages will be built and installed:
    embree3[core]:arm64-osx -> 3.13.5#1
Detecting compiler hash for triplet arm64-osx...
Restored 0 package(s) from /Users/xxx/.cache/vcpkg/archives in 552.209 us. Use --debug to see more details.
Installing 1/1 embree3:arm64-osx...
Building embree3[core]:arm64-osx...
warning: -- Using community triplet arm64-osx. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/xxx/vcpkg/triplets/community/arm64-osx.cmake
-- Downloading https://github.com/embree/embree/archive/v3.13.5.tar.gz -> embree-embree-v3.13.5.tar.gz...
-- Extracting source /Users/xxx/vcpkg/downloads/embree-embree-v3.13.5.tar.gz
-- Applying patch no-runtime-install.patch
-- Using source at /Users/xxx/vcpkg/buildtrees/embree3/src/v3.13.5-373c034cde.clean
-- Found external ninja('1.11.1').
-- Configuring arm64-osx-dbg
-- Configuring arm64-osx-rel
CMake Warning at installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:314 (message):
  The following variables are not used in CMakeLists.txt:

      EMBREE_STATIC_RUNTIME

  Please recheck them and remove the unnecessary options from the
  `vcpkg_cmake_configure` call.

  If these options should still be passed for whatever reason, please use the
  `MAYBE_UNUSED_VARIABLES` argument.
Call Stack (most recent call first):
  ports/embree3/portfile.cmake:48 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)

-- Building arm64-osx-dbg
-- Building arm64-osx-rel
-- Installing: /Users/xxx/vcpkg/packages/embree3_arm64-osx/share/embree3/copyright
-- Performing post-build validation
Stored binary cache: "/Users/xxx/.cache/vcpkg/archives/45/451e160d9ddc9fc3c2a29c842d832645b97effca67432e53dda4747074a51225.zip"
Elapsed time to handle embree3:arm64-osx: 5.548940125 min
Total install time: 5.594444395833333 min
The package embree3 provides CMake targets:

    find_package(embree 3 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE embree)

### PR DESCRIPTION
**Pull request by autoantwort**
Fixes #30541

### CODE REVIEW

File: ports/embree3/vcpkg.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "embree3",
   "version": "3.13.5",
-  "port-version": 1,
+  "port-version": 2,
   "description": "High Performance Ray Tracing Kernels.",
   "homepage": "https://github.com/embree/embree",
   "license": "Apache-2.0",
-  "supports": "!arm",
+  "supports": "!(arm & !osx)",
```
Comment (MonicaLiu0311) [inline ports/embree3/vcpkg.json:8]: Maybe you meant `"support": "osx & !arm"`?

Comment (Nosenzor) [reply ports/embree3/vcpkg.json:8]: That's the same

Comment (dg0yt) [reply ports/embree3/vcpkg.json:8]: No, it isn't the same. 
`osx & !arm` is the same as `osx & x64`. And this too restrictive.

The alternative formulation is: `!arm | osx`.
While the field is named `supports`, the guideline is to only exclude what is known to be unsupported. The proposed condition captures this well: It allows everything which is not arm, plus everything which is osx.

**Comment by MonicaLiu0311** [comment]
Except for avx512, other features are tested successfully in the following triplets:
```
x86-windows
x64-windows
x64-windows-static
```

All features are tested successfully in the following triplet:
x64-osx

File: ports/embree3/vcpkg.json (hunk 1 above)
Comment (autoantwort) [reply ports/embree3/vcpkg.json:8]: `!arm | osx` is maybe easier to understand

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**